File: Decorato.java/src/com/parttern/decorator/Flan.java
package com.parttern.decorator;

public class Flan extends ToppingDecorator{
    private double price = 0.5;
    public Flan(MilkTea innerMilkTea) {
        super(innerMilkTea);
    }

    @Override
    public double cost() {
        return this.price + milkTea.cost();
    }

    @Override
    public String doMilkTea() {
        String type = milkTea.doMilkTea();
        return type + addFlan();
    }

    private String addFlan() {
        return " ,Bánh Flan";
    }
}

File: Decorator/Flan.cs
namespace Decorator {
    internal class Flan : ToppingDecorator {
        private readonly double price = 0.5;
        public Flan(MilkTea milkTea) : base(milkTea) {
        }
        public override double Cost() {
            return price + milkTea.Cost();
        }

        public override string DoMilkTea() {
            return milkTea.DoMilkTea() + AddFlan();
        }

        private string AddFlan() {
            return ", Bánh Flan";
        }
    }
}
